src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -229,6 +229,8 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
                     break;
             }
 
+            Debug.Assert(!_http3Enabled || _http2Enabled, "We shouldn't support H3 while disabling H2.");
```

Comment:
Why is that the case?

Reply:
I added the assert since we were relying on that fact with how the conditions were set up (init logic for H3 was behind `if (http2Enabled)`).
Looking at it again, I don't see why we were duplicating the code in the first place - I dropped the assert.